### ISSUE TITLE
Rubocop incorrectly suggests: Style/TernaryParentheses: Omit parentheses for ternary conditions

### ISSUE THREAD

**Issue by jspencer-cmm** [rubocop-hq/rubocop#7864]
When a method call is used as the first value into the ternary operator, rubocop suggests dropping the parentheses around the method call. This breaks the logic because the method is not called first before evaluating the ternary operator, it is instead called on the result of the ternary operator.

Tested using rubocop version 0.81.0.

--------

## Expected behavior

Rubocop should not suggest removing the parentheses in this situation.

## Actual behavior

Rubocop suggests removing the parentheses around the method invocation.

## Steps to reproduce the problem

```ruby
def true?(_item)
  true
end

# this is correct
puts ((true? :foo) ? :bar : :baz) # :bar

# rubocop suggests the following:
# C: Style/TernaryParentheses: Omit parentheses for ternary conditions.
# puts ((true? :foo) ? :bar : :baz) # :bar
#       ^^^^^^^^^^^^^^^^^^^^^^^^^^

# but this is incorrect, now it returns true instead of :bar
puts (true? :foo ? :bar : :baz) # true
```

## RuboCop version

```
[root@f07a16c8f099 lib]# rubocop -V
0.81.0 (using Parser 2.7.1.0, running on ruby 2.4.6 x86_64-linux-gnu)
```

**Comment by tejasbubane** [comment]
Actually the steps to reproduce would be:
```ruby
puts (true? :foo) ? :bar : :baz

# rubocop suggests the following:
test.rb:5:6: C: Style/TernaryParentheses: Omit parentheses for ternary conditions.
puts (true? :foo) ? :bar : :baz
     ^^^^^^^^^^^^^^^^^^^^^^^^^^
```

Although using `(true? :foo)` itself is an offense but that is a job of `Lint/ParenthesesAsGroupedExpression`. So `Style/TernaryParentheses` should not register offense in this case.

And the accepted way to write above expression is:
```ruby
puts true?(:foo) ? :bar : :baz
```

**Comment by tejasbubane** [comment]
On another look at the cop, you can use `EnforceStyle: require_parentheses_when_complex` for the cop and have it not register offense for above code.

**Comment by jonas054** [comment]
Not a bug and no activity in the issue for a month. Closing.
